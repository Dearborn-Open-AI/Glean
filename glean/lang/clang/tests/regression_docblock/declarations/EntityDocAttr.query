query: docmarkup.EntityDocAttr.7 _
transform: [gensort, []]
