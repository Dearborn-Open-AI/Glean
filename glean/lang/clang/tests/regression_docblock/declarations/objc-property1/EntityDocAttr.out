[
  "@generated",
  {
    "key": {
      "cxx": {
        "decl": {
          "objcContainer": {
            "key": {
              "id": { "implementation": { "key": "A" } },
              "source": {
                "file": { "key": "test.m" },
                "lineBegin": 33,
                "columnBegin": 1,
                "lineEnd": 37,
                "columnEnd": 1
              }
            }
          }
        }
      }
    },
    "value": [
      {
        "key": {
          "key": { "key": "testname" },
          "value": "declarations/objc-property1"
        }
      },
      { "key": { "key": { "key": "type" }, "value": "implementation A" } }
    ]
  },
  {
    "key": {
      "cxx": {
        "decl": {
          "objcContainer": {
            "key": {
              "id": { "interface_": { "key": "A" } },
              "source": {
                "file": { "key": "test.m" },
                "lineBegin": 13,
                "columnBegin": 1,
                "lineEnd": 27,
                "columnEnd": 4
              }
            }
          }
        }
      }
    },
    "value": [
      {
        "key": {
          "key": { "key": "testname" },
          "value": "declarations/objc-property1"
        }
      },
      { "key": { "key": { "key": "type" }, "value": "interface A" } }
    ]
  },
  {
    "key": {
      "cxx": {
        "decl": {
          "objcProperty": {
            "key": {
              "name": { "key": "i" },
              "container": { "interface_": { "key": "A" } },
              "type": { "key": "int" },
              "isInstance": true,
              "isOptional": false,
              "isReadOnly": false,
              "isAtomic": false,
              "source": {
                "file": { "key": "test.m" },
                "lineBegin": 25,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 35
              }
            }
          }
        }
      }
    },
    "value": [
      {
        "key": {
          "key": { "key": "testname" },
          "value": "declarations/objc-property1"
        }
      },
      { "key": { "key": { "key": "type" }, "value": "property i" } }
    ]
  },
  {
    "key": {
      "cxx": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": { "name": { "key": "_i" }, "scope": { "global_": { } } }
              },
              "type": { "key": "int" },
              "kind": { "ivar": { "synthesize": false } },
              "source": {
                "file": { "key": "test.m" },
                "lineBegin": 18,
                "columnBegin": 3,
                "lineEnd": 18,
                "columnEnd": 8
              }
            }
          }
        }
      }
    },
    "value": [
      {
        "key": {
          "key": { "key": "testname" },
          "value": "declarations/objc-property1"
        }
      },
      { "key": { "key": { "key": "type" }, "value": "A::_i" } }
    ]
  }
]